### PR TITLE
[BUGFIX] Check for array-key existence

### PR DESCRIPTION
**Pull request by lsascha**
Should prevent
```
Warning: Undefined array key
```
warnings

Log:
```´
[ TYPO3\CMS\Core\Error\Exception ]
Warning: Undefined array key "foreign_table" in /var/www/html/web/typo3conf/ext/file_variants/Configuration/TCA/Overrides/manipulation.php line 22

[ TYPO3\CMS\Core\Error\Exception ]
Warning: Undefined array key "languageField" in /var/www/html/web/typo3conf/ext/file_variants/Configuration/TCA/Overrides/manipulation.php line 30

[ TYPO3\CMS\Core\Error\Exception ]
Warning: Undefined array key "sys_language_uid" in /var/www/html/web/typo3conf/ext/file_variants/Configuration/TCA/Overrides/manipulation.php line 32
```

### CODE REVIEW

**Comment by mbrodala** [comment]
Can you simplify this using the null coalescing operator (`??`)? We don't require a PHP version specifically but at minimum TYPO3v10 which itself requires PHP 7.2+

**Comment by mbrodala** [comment]
@lsascha Can you adjust this as requested?

**Comment by chrisliquidlight** [comment]
@mbrodala @lsascha Is there any update on this PR, or anything I can contribute? The extension is currently incompatible with PHP 8 until this is added

File: Configuration/TCA/Overrides/manipulation.php [hunk 1]
```diff
@@ -19,20 +19,24 @@
         //   if ($recordService->isFalConsumingTable($table)) {
         // streamline language sync for all FAL fields
         foreach ($config['columns'] as $fieldName => $fieldConfig) {
-            if ($fieldConfig['config']['foreign_table'] === 'sys_file_reference') {
+            if (isset($fieldConfig['config']['foreign_table']) && $fieldConfig['config']['foreign_table'] === 'sys_file_reference') {
```
Comment (mbrodala) [inline Configuration/TCA/Overrides/manipulation.php:22]:
```suggestion
            if ($fieldConfig['config']['foreign_table'] ?? null === 'sys_file_reference') {
```

File: Configuration/TCA/Overrides/manipulation.php [hunk 2]
```diff
@@ -19,20 +19,24 @@
...
+        if (isset($GLOBALS['TCA'][$table]['ctrl']['languageField'])) {
+            $languageField = $GLOBALS['TCA'][$table]['ctrl']['languageField'];
+            if (isset($config['columns'][$languageField])) {
+                $fieldConfig = $config['columns'][$languageField]['config'];
+                if (isset($fieldConfig['items'])) {
+                    foreach ($fieldConfig['items'] as $index => $item) {
+                        if ((int)$item[1] === -1) {
+                            unset($GLOBALS['TCA'][$table]['columns'][$languageField]['config']['items'][$index]);
+                        }
+                    }
```
Comment (mbrodala) [inline Configuration/TCA/Overrides/manipulation.php:39]:
```suggestion
        $languageField = $GLOBALS['TCA'][$table]['ctrl']['languageField'] ?? null;
        $fieldConfig = $config['columns'][$languageField]['config'] ?? [];
        if ($languageField && isset($fieldConfig['items'])) {
            foreach ($fieldConfig['items'] as $index => $item) {
                if ((int)$item[1] === -1) {
                    unset($GLOBALS['TCA'][$table]['columns'][$languageField]['config']['items'][$index]);
```

**Review by mbrodala — CHANGES_REQUESTED** [review]

**Comment by mbrodala** [comment]
@chrisliquidlight Done now.

**Comment by chrisliquidlight** [comment]
@mbrodala Thank you!

**Pull request merged**